app/src/main/java/com/yacgroup/yacguide/PartnersActivity.kt

```diff
@@ -144,56 +133,53 @@ class PartnersActivity : AppCompatActivity() {
         }
 
         for (partner in sortedPartners) {
-            val partnerName = partner.name.orEmpty()
             val innerLayout = RelativeLayout(this)
             innerLayout.setBackgroundColor(Color.WHITE)
 
-            val checkBox = CheckBox(this)
-            checkBox.text = partnerName
-            if (_selectedPartnerIds.contains(partner.id)) {
-                checkBox.isChecked = true
+            CheckBox(this).let {
+                it.text = partner.name.orEmpty()
+                if (_selectedPartnerIds.contains(partner.id)) {
```

Comment:
`false` seems to be the default value because I didn't change this particular code line and it worked before, too. However, the suggested solution is safer and needs only one code line.